### PR TITLE
Odim 259 - Adding logrus support to odimra 

### LINKED ISSUE #259: migrate logging to logrus
migrate from go lang std. log package to logrus. Also 
1) introduce log levels
2) change calls to fmt.* to log.* except where fmt funcs are used to format strings

### PR DESCRIPTION
**Pull request by Bharath-KKB**
Address issue 259
closes #259

### CODE REVIEW

**Review by a-ajith — APPROVED** [review]

File: lib-messagebus/datacommunicator/kafkacomm.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func KafkaConnect(kp *KafkaPacket, messageQueueConfigPath string) error {
 	// Creation of TLS Config and Dialer
 	tls, e := TLS(mq.KAFKACertFile, mq.KAFKAKeyFile, mq.KAFKACAFile)
 	if e != nil {
-		log.Printf("%v", e)
+		log.Error("%v" + e.Error())
```
Comment (skhan-anjum) [inline lib-messagebus/datacommunicator/kafkacomm.go:115]: I think "%v" is a formatting directive. If used with '+', Go will consider it as a string. We should either use it with logrus.Errorf or we should remove it. Please modify this at all the places if the need be.

File: lib-rest-client/pmbhandle/rest.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func ContactPlugin(url, method, token string, odataID string, body interface{},
 	}
 
 	if resp.StatusCode >= 300 {
-		log.Printf("warning: got %v, while fetching %v with method %v", resp.Status, url, method)
+		log.Warn("got %v, while fetching %v with method %v" + resp.Status + url + method)
```
Comment (a-ajith) [inline lib-rest-client/pmbhandle/rest.go:66]: log.Warn("got %v, while fetching %v with method %v" + resp.Status + url + method)
can be changed to
log.Warnf("got %v, while fetching %v with method %v", resp.Status, url, method)

File: plugin-redfish/rfphandler/subscribeEvents.go [hunk 3]
```diff
@@ -102,31 +103,31 @@ func deleteMatchingSubscriptions(device *rfputilities.RedfishDevice) {
...
 	if err != nil {
-		log.Println(err.Error())
+		log.Error(err.Error())
 		return
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		errorMessage := fmt.Sprintf("error: while getting subscription details for URI: %v got %v", device.Location, resp.StatusCode)
-		log.Println(errorMessage)
+		errorMessage := fmt.Sprintf("Unable to get subscription details for URI: " + device.Location + " got " + strconv.Itoa(resp.StatusCode))
```
Comment (a-ajith) [inline plugin-redfish/rfphandler/subscribeEvents.go:118]: fmt.Sprintf converts and concatenates the passed arguments into a string. So we can avoid the use of 'strconv.Itoa'

File: lib-messagebus/datacommunicator/kafkacomm.go (hunk 1 above)
Comment (Bharath-KKB) [reply lib-messagebus/datacommunicator/kafkacomm.go]: Yes. removed it

File: lib-rest-client/pmbhandle/rest.go (hunk 2 above)
Comment (Bharath-KKB) [reply lib-rest-client/pmbhandle/rest.go]: Will change in future

File: plugin-redfish/rfphandler/subscribeEvents.go (hunk 3 above)
Comment (Bharath-KKB) [reply plugin-redfish/rfphandler/subscribeEvents.go]: I dont like that but changed it

**Review by shivacharanms — APPROVED** [review]

**Review by skhan-anjum — APPROVED** [review]

**Review by jeevan-kamkar — COMMENTED** [review]
two different version logrus are refereed in go.mod file we can pin this to a single  version
`github.com/sirupsen/logrus v1.4.2`
`github.com/sirupsen/logrus v1.7.0`

we can pin it to  one of the version using this command 
`go get github.com/sirupsen/logrus@"version"`

**Review by jeevan-kamkar — APPROVED** [review]
LGTM

**Pull request merged**